manifest-build-tools/lib/RepositoryOperator.py

```diff
@@ -0,0 +1,417 @@
+                credentials = self._git_credentials
+        else:
+            self._git_credentials = credentials
+        for url_cred_pair in credentials:
+            url, cred = url_cred_pair.split(',')
+            self.git.add_credential_from_variable(url, cred)
+
+    def set_git_dryrun(self, dryrun):
+        self.git.set_dryrun(dryrun)
+    
+    def set_git_verbose(self, verbose):
+        self.git.set_verbose(verbose)
+
+    def set_git_executable(self, executable):
+        self.git.set_excutable(excutable)
+
+
+    @staticmethod
+    def print_command_summary(name, results):
+        """
+        Print the results of running commands.
+          first the command line itself
+            and the error code if it's non-zero
+          then the stdout & stderr values from running that command
+
+        :param name:
+        :param results:
+        :return: True if any command exited with an error condition
+        """
+
+        error_found = False
+
+        print "============================"
+        print "Command output for {0}".format(name)
+
+        if 'commands' in results[name]:
+            commands = results[name]['commands']
+            for command in commands:
+                for key in ['command', 'stdout', 'stderr']:
+                    if key in command:
+                        if command[key] != '':
+                            print command[key]
+                        if key == 'command':
+                            if command['return_code'] != 0:
+                                error_found = True
+                                print "EXITED: {0}".format(command['return_code'])
+
+        return error_found
+
+    def clone_repo_list(self, repo_list, dest_dir, jobs=1):
+        """
+        check out repository to dest dir based on repo list
+        :param repo_list: a list of repository entry which should contain:
+                          'repository': the url of repository, it is required
+                          'branch': the branch to be check out, it is optional
+                          'commit-id': the commit id to be reset, it is optional
+        :param dest_dir: the directory where repository will be check out
+        :param jobs: Number of parallel jobs to run
+        :return:
+        """
+        cloner = RepoCloner(jobs)
+        if cloner is not None:
+            for repo in repo_list:
+                data = {'repo': repo,
+                        'builddir': dest_dir,
+                        'credentials': self._git_credentials
+                       }
+
+                cloner.add_task(data)
+            cloner.finish()
+            results = cloner.get_results()
+
+            error = False
+            for name in results.keys():
+                error |= self.print_command_summary(name, results)
+
+            if error:
+                raise RuntimeError("Failed to clone repositories")
+
+    
+    def clone_repo(self, repo_url, dest_dir, repo_commit="HEAD"):
```

Comment:
The class RepoCloner support add several tasks and clone repositories in parallel.
An example of the RepoCloner is the function clone_repo_list